autogen/agentchat/contrib/multimodal_conversable_agent.py

```diff
@@ -85,6 +82,14 @@ def _message_to_dict(message: Union[Dict, List, str]) -> Dict:
             return {"content": message}
         if isinstance(message, dict):
             assert "content" in message, "The message dict must have a `content` field"
+            # GPT-4 does not support function calls yet.
```

Comment:
Do you think it is better to store the raw messages and only transform them before sending it to the LLM?

Reply:
What I meant is doing the transformation of using None inside `generate_oai_reply` rather than overriding the parent's private method.

Reply:
The message content inside is updated anyway. For instance, the images are converted from URL to PIL or base64 content in this function. 

The (future) storage/logging layer should also be able to handle different underlying message content? 

Let's say the user sent a 4 GB video to the agent, and we don't modify the message here in the `_message_to_dict` function. Then, would the storage feature just store the 4 GB raw video in the log?

Reply:
Yes I believe so. The handling of caching and references should be internal logic of the storage system rather than exposing it externally.

Reply:
Message data has 3 states: transit, compute and storage. Preparing the logic for preparing message data for compute shouldn't affect how it is stored. Otherwise the stored data cannot be used by other compute with different requirements

Reply:
Resolving this for now, we can discuss overall design later.